### ISSUE TITLE
marshaling a duplicate should consume Hizdahr cost reduction

### ISSUE THREAD

**Issue by zacchiro** [throneteki/throneteki#2278]
**Scenario**: trigger Hizdahr, marshal a duplicate, marshal a card that cost 3 or less.
**Result**: the last card will be marshaled for free.
**Correct behavior**: the last card should be paid its full cost, because Hizdahr reduction has been wasted on marshaling the duplicate.

Rationale: even if marshaling a duplicate doesn't count as marshaling a card of any specific type (it's just "marshaling a duplicate card"), it is still marshaling *a card* and Hizdahr reduces that specifically.

I haven't checked if this is a more general issue affecting other reducers of it is Hizdahr-specific.

**Comment by ystros** [comment]
Good catch. It should be Hizdahr specific, since he is the only card currently that reduces "marshaling a card" with no characteristic requirements. #2273 fixes reducing events brought out of shadow, but a fix for this will probably need to be more in-depth, as reducers aren't checked at all for dupes currently.

**Comment by mplain** [comment]
I don't think this is right. Marshaling a duplicate has *no cost*, so cost modifiers do not apply.

Similar scenario is using Ocean Road and then marshaling a duplicate on an out-of-faction character - the discount should not be wasted.

Note that while a facedown card (with shadows) is considered to be neither in-faction not out-of-faction nor neutral, this is not true for duplicates.

**Comment by zacchiro** [comment]
On Sun, Oct 14, 2018 at 09:56:39AM -0700, mplain wrote:
> I don't think this is right. Marshaling a duplicate has *no cost*, so
> cost modifiers do not apply.
> 
> Similar scenario is using Ocean Road and then marshaling a duplicate
> on an out-of-faction character - the discount should not be wasted.

Do you have a reference backing any of these?

**Comment by mplain** [comment]
Deleted my comment because although cost reduction should not apply to marshaling a duplicate (because marshaling a duplicate has no cost), still the card marshaled after a duplicate will not be "the next card you marshal" after you trigger Hizdahr, so his discount will not apply to it either. So in the end yes, Hizdahr's discount should be "wasted" if you marshal a dupe.

Regarding Ocean Road - I think i'll need to ask Danny about that.

**Comment by zacchiro** [comment]
On Sun, Oct 14, 2018 at 11:24:57AM -0700, mplain wrote:
> A duplicate is not a 0-cost card. Marshaling a duplicate has *no cost*.

Sure. But so what? The counter argument is that if you use a discount
for something that has 0 cost it's your loss. You'd need an exception
saying that reducing something that costs 0 doesn't use the reduction,
and I'm still waiting to see that exception ruled.

**Comment by mplain** [comment]
I do not understand your argument. I keep saying that marshaling a duplicate has no cost, and you keep saying that it has 0 cost, but we know that "no cost" and "0 cost" are two very different things.

If you marshal something that costs 0, then in step 3 you apply cost modifiers and go to 0-X= -X which is cnsidered to be 0, because values can't go negative.

If something has no cost, then there is nothing to apply cost modifiers to.
